diff --git a/HDBC.cabal b/HDBC.cabal
index 9e31c64..3d36452 100644
--- a/HDBC.cabal
+++ b/HDBC.cabal
@@ -35,7 +35,7 @@ library
   if flag(splitBase)
     Build-Depends: base>=3 && <5, old-time, bytestring, containers
     if flag(minTime15)
-      Build-Depends: time >= 1.5 && < 1.14
+      Build-Depends: time >= 1.5 && < 1.15
       CPP-Options: -DMIN_TIME_15
     else
       Build-Depends: time >= 1.2 && < 1.5, old-locale
